### PR TITLE
Migrate to flutter 2

### PR DESCRIPTION
**Pull request by jonataslaw**
Hi, I noticed that the project became incompatible with one of our projects after the Flutter 2 update, so I migrated it

### CODE REVIEW

File: lib/src/posthog.dart [hunk 1]
```diff
@@ -30,9 +28,9 @@ class Posthog {
   }
 
   Future<void> capture({
-    @required String eventName,
-    Map<String, dynamic> properties,
-    Map<String, dynamic> options,
+    required String eventName,
+    required Map<String, dynamic> properties,
```
Comment (yakkomajuri) [inline lib/src/posthog.dart:32]: this seems to have diverged from lib/src/posthog_method_channel.dart where properties are correctly assigned as optional

File: example/lib/main.dart [hunk 2]
```diff
@@ -61,49 +61,52 @@ class MyApp extends StatelessWidget {
...
             Spacer(),
             Center(
-              child: FlatButton(
+              child: TextButton(
                 child: Text('Enable'),
                 onPressed: () async {
-                  await Posthog.enable();
-                  Posthog.capture(eventName: 'Enabled capturing events!');
+                  await Posthog().enable();
+                  Posthog().capture(
```
Comment (yakkomajuri) [inline example/lib/main.dart:98]: Is this standard Dart indentation? (I don't know so I'm truly asking)

**Review by yakkomajuri — COMMENTED** [review]
Hey, thank you so much for doing this! We don't have a dedicated Dart/Flutter dev hence it took a while to get around to this. 

Left a few minor comments but I'm still testing this out

**Comment by yakkomajuri** [comment]
@jonataslaw this worked for me and all the changes make sense! Thanks again for doing this. A few minor comments otherwise happy to merge.

File: example/lib/main.dart (hunk 2 above)
Comment (jonataslaw) [reply example/lib/main.dart:98]: Yeah, it's pretty odd for those who are used to other languages, but it's the default, I even used dart format in this file before saving the changes

File: lib/src/posthog.dart (hunk 1 above)
Comment (jonataslaw) [reply lib/src/posthog.dart:32]: You're right, I'm going to make it nullable instead of required

**Comment by jonataslaw** [comment]
I just changed properties to nullable to be consistent with posthog_method_channel now

**Review by yakkomajuri — APPROVED** [review]
Thanks again! Will get this in but I don't have access to publish a new version, will look to get access to this internally

**Comment by fuziontech** [comment]
lib pushed to pub.dev! 
https://pub.dev/packages/posthog_flutter

Available version is now 2.0.0

**Pull request merged**